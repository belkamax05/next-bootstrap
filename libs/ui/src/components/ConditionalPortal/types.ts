import { PropsWithChildren } from 'react';

export type ConditionalPortalProps = PropsWithChildren<{
  condition: boolean;
  portalNodeId: string;
}>;
