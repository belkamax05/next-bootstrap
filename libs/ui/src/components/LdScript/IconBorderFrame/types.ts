import { ReactNode } from 'react';

export interface IconBorderFrameProps {
  size: number;
  svgSize?: number;
  before?: ReactNode;
  layout?: ReactNode;
  className?: string;
  backgroundColor?: string;
}
