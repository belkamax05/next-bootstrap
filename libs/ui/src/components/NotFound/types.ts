export interface NotFoundProps {
  title?: string;
  showIcon: boolean;
  description: string;
  at: 'error-layout' | 'context-search-menu' | 'marketing-preference';
}
