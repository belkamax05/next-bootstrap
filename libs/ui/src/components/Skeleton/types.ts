export interface SkeletonProps {
  animated?: boolean;
  borderRadius?: string;
  count?: number;
  color?: string;
  highlightColor?: string;
  height?: string;
  space?: string;
  width?: string;
  id: string;
}
