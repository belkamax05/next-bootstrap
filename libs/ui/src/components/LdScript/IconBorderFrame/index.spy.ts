import * as IconBorderFrame from '.';

const IconBorderFrameSpy = jest.spyOn(IconBorderFrame, 'default');

export default IconBorderFrameSpy;
