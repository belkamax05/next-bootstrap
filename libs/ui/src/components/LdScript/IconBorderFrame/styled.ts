import excludeProps from '@/utils/helpers/excludeProps';
import Box from '@mui/material/Box';
import { styled } from '@mui/material/styles';
import { IconBorderFrameProps } from './types';

const name = 'IconBorderFrame';

export const StyledRoot = styled(Box, {
  name,
  slot: 'root',
  shouldForwardProp: excludeProps(['size', 'svgSize', 'backgroundColor']),
})<Pick<IconBorderFrameProps, 'size' | 'svgSize' | 'backgroundColor'>>(
  ({ theme, size, svgSize, backgroundColor }) => ({
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    minHeight: size,
    minWidth: size,
    position: 'relative',
    border: `1px solid ${theme.palette.copy.main}`,
    borderRadius: 4,
    padding: 2,
    backgroundColor,
    '& > div > div > img': {
      display: 'none',
    },
    '& svg': {
      width: svgSize,
      height: svgSize,
    },
  }),
);
