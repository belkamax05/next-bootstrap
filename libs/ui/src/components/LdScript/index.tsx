'use server';
import { SchemaOrgBase } from '@/types';

export type LdScriptProps<TData extends SchemaOrgBase = SchemaOrgBase> = {
  data: TData;
};

const LdScript = <TData extends SchemaOrgBase>({ data }: LdScriptProps<TData>) => {
  if (!data) return null;
  const dataStr = JSON.stringify(data || {});
  return (
    <script
      type="application/ld+json"
      data-testid="schema-org-script"
      data-schema-type={data['@type']}
      dangerouslySetInnerHTML={{ __html: dataStr }}
    />
  );
};

export default LdScript;
