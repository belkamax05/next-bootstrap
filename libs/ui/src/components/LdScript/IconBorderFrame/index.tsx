import { StyledRoot } from './styled';
import { IconBorderFrameProps } from './types';

const IconBorderFrame = ({
  layout,
  backgroundColor = 'transparent',
  before,
  className,
  size,
  svgSize,
}: IconBorderFrameProps) => {
  return (
    <StyledRoot
      className={className}
      size={size}
      svgSize={svgSize}
      backgroundColor={backgroundColor}
    >
      {before}
      {layout}
    </StyledRoot>
  );
};
export default IconBorderFrame;
