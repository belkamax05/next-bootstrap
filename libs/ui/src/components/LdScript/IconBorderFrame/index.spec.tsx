import IconImage from '@/component-library/components/image/IconImage';
import { pdpAttributeColorWheelTestIds } from '@/component-library/pages/PdpV2/components/PdpAttributeColorWheel';
import { render } from '@/ui/utils/test-utils';
import IconBorderFrame from '.';
import { IconBorderFrameProps } from './types';

const setup = (props?: Partial<IconBorderFrameProps>) =>
  render(
    <IconBorderFrame
      size={20}
      before={<div data-testid={pdpAttributeColorWheelTestIds.icon}>PdpAttributeColorWheel</div>}
      layout={<IconImage name="imageName" src="imageSrc" alt="imageAlt" />}
      {...props}
    />,
  );

describe('IconBorderFrame', () => {
  it('should match snapshot', () => {
    const { baseElement } = setup();

    expect(baseElement).toMatchSnapshot();
  });

  it('should render image', () => {
    const { getByAltText } = setup();

    expect(getByAltText('imageAlt')).toBeInTheDocument();
  });

  it('should render colour wheel', () => {
    const { getByTestId } = setup();

    expect(getByTestId(pdpAttributeColorWheelTestIds.icon)).toBeInTheDocument();
  });

  it('should not render colour wheel', () => {
    const { queryByTestId } = setup({ before: null });

    expect(queryByTestId(pdpAttributeColorWheelTestIds.icon)).not.toBeInTheDocument();
  });
  it('show image if src and name is available', () => {
    const { getByAltText } = setup();

    expect(getByAltText('imageAlt')).toBeVisible();
  });

  it('should show svg if svg is present', () => {
    const { getByText, baseElement } = setup({
      before: null,
      layout: <>SVG Content</>,
    });

    expect(baseElement).toMatchSnapshot();
    expect(getByText('SVG Content')).toBeVisible();
  });
});
