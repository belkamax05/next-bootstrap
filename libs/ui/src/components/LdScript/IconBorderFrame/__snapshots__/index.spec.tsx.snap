// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconBorderFrame should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1rcu5ey-IconBorderFrame-root"
    >
      <div
        data-testid="colour-wheel-icon"
      >
        PdpAttributeColorWheel
      </div>
      <img
        alt="imageAlt"
        class="css-xkwt5p-IconImage-image"
        height="40"
        placeholder="blur"
        src="imageSrc"
        title="imageName"
        width="40"
      />
    </div>
  </div>
</body>
`;

exports[`IconBorderFrame should show svg if svg is present 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1rcu5ey-IconBorderFrame-root"
    >
      SVG Content
    </div>
  </div>
</body>
`;
